src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -210,6 +214,60 @@ def shutdown(self):
         self.log.info('Stopping pg_autoscaler')
         self._shutdown.set()
 
+    def check_pool_activity(check_time, pool_stats, pool_id, final_pg_target, old_values):
+        total = (int(pool_stats.get("rd", 0)) + int(pool_stats.get("wr", 0)))
+        if pool_id in old_values.keys():
+            if check_time == 1:
+                if old_values[pool_id]['times'] == total:
+                    old_values[pool_id]['unused'] += 1
+                    old_values[pool_id]['add'] = 0
+                    old_values[pool_id]['actual_add'] = 0
+                    if final_pg_target - old_values[pool_id]['unused'] < PG_NUM_MIN:
+                        old_values[pool_id]['unused'] -= 1
+                    else:
+                        for k1 in old_values:
+                            if old_values[k1]['unused'] == 0 and pool_id != k1:
+                                old_values[k1]['add'] += 1
+                else:
+                    if old_values[pool_id]['unused'] != 0:
+                        old_values[pool_id]['unused'] = 0
+                        old_values[pool_id]['actual_unused'] = 0
+                    else:
+                        for k2 in old_values:
+                            while old_values[k2]['add'] != 0 and old_values[pool_id]['unused'] != 0:
+                                if k2 != old_values[pool_id]:
+                                    old_values[k2]['add'] -= 1
+                                    old_values[pool_id]['unused'] -= 1
+            else:
+                if old_values[pool_id]['add'] - old_values[pool_id]['actual_add'] >= \
+                        nearest_power_of_two(final_pg_target, "High") - final_pg_target:
+                    for k3 in old_values:
+                        while old_values[pool_id]['actual_add'] < old_values[pool_id]['add'] and \
+                                old_values[k3]['unused'] > old_values[k3]['actual_unused']:
+                            if k3 != old_values[pool_id]:
+                                old_values[pool_id]['actual_add'] += 1
+                                old_values[k3]['actual_unused'] += 1
+#                        final_pg_target += old_values[pool_id]['actual_add']
+                elif int(old_values[pool_id]['unused']) - int(old_values[pool_id]['actual_unused']) > \
+                        final_pg_target - nearest_power_of_two(final_pg_target, "Low"):
+                    for k3 in old_values:
+                        while old_values[k3]['actual_add'] < old_values[k3]['add'] and \
+                                old_values[pool_id]['unused'] > old_values[pool_id]['actual_unused']:
+                            if k3 != old_values[pool_id]:
```

Comment:
Same as above, should we skip the entire inner while loop if k3 != old_values[pool_id]?

Reply:
Try getting rid of elif above